zerver/tests/test_narrow.py

```diff
@@ -504,6 +519,35 @@ def dr_emails(dr):
             for message in result["messages"]:
                 self.assertEqual(dr_emails(message['display_recipient']), emails)
 
+    def test_get_messages_with_narrow_huddle_with(self):
+        # type: () -> None
+        """
+        A request for old messages with a narrow by huddle-with only returns
+        group-private conversations with that user.
+        """
+        me = 'hamlet@zulip.com'
+
+        def dr_emails(dr):
+            # type: (Union[Text, List[Dict[str, Any]]]) -> List[Any]
+            assert isinstance(dr, list)
+            emails = []
+            for r in dr:
+                emails.append(r['email'])
+            return emails
+
+        self.send_message(me, ['iago@zulip.com', 'cordelia@zulip.com', 'othello@zulip.com'], Recipient.HUDDLE)
+        self.send_message(me, ['cordelia@zulip.com', 'othello@zulip.com'], Recipient.HUDDLE)
+        self.send_message(me, 'cordelia@zulip.com', Recipient.PERSONAL)
+        self.send_message(me, ['iago@zulip.com', 'othello@zulip.com'], Recipient.HUDDLE)
+        self.send_message('cordelia@zulip.com', ['iago@zulip.com', 'othello@zulip.com'], Recipient.HUDDLE)
+
+        self.login(me)
+        narrow = [dict(operator='huddle-with', operand='cordelia@zulip.com')]
+        result = self.get_and_check_messages(dict(narrow=ujson.dumps(narrow)))
+        for message in result["messages"]:
+            self.assertIn(me, dr_emails(message['display_recipient']))
+            self.assertIn('cordelia@zulip.com', dr_emails(message['display_recipient']))
```

Comment:
`self.send_message` returns a message ID, so might be simplest to just create 2 lists -- `matching_message_ids` and `nonmatching_message_ids` and check that each list is correctly reflected in the result.